### PR TITLE
Submitted obsidian-tesseract-ocr plugin

### PR DESCRIPTION
**Pull request by kaffarell**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/kaffarell/obsidian-tesseract-ocr

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
["fundingUrl": "https://github.com/kaffarell",](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/dc6f53bee193438734815ca1b9643323c693a28f/manifest.json#L9)
This is meant for links to services like _buy me a coffee_, _GitHub Sponsors_ etc, if you don't have such a link omit this.

["isDesktopOnly": false](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/dc6f53bee193438734815ca1b9643323c693a28f/manifest.json#L10)
Your plugin depends on NodeJS only API's, which is only available on desktop, set this to `true`.

[let content = await this.app.vault.adapter.read(file.path)](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/dc6f53bee193438734815ca1b9643323c693a28f/src/main.ts#L58)
Use the vault API wherever possible: `this.app.vault.cachedRead(file)`

[private getAllFiles(): TAbstractFile[] {](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/dc6f53bee193438734815ca1b9643323c693a28f/src/main.ts#L111-L117)
This function can be replaced with a call to `this.app.vault.getAllLoadedFiles()`

[private isImage(fileName: string): boolean {](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/dc6f53bee193438734815ca1b9643323c693a28f/src/main.ts#L134-L140)
Instead of this, do something like this:
```ts
private isImage(file: TAbstractFile) {
    return file instanceof TFile && ['jpg', 'png', 'jpeg'].includes(file.extension);
}
```

I would also recommend making the location of the tesseract executable configurable, while having it in the path works most of the time, there are cases where this does not work.

**Comment by kaffarell** [comment]
Hey @joethei thanks for the guidance! Made the changes and released a new version.

File: community-plugins.json [hunk 1]
```diff
@@ -6337,5 +6337,12 @@
        "description": "Obsidian Unhelper. 2023 April Fool's plugin brought to you by Obsidian.",
        "author": "Obsidian",
        "repo": "ericaxu/Gemmy"
+    },
+    {
+       "id": "image-ocr",
+       "name": "Image OCR",
+       "description": "Runs ocr on images and pastes result in description.",
```
Comment (liamcain) [inline community-plugins.json:6344]:
```suggestion
       "description": "Runs OCR on images and pastes result in description.",
```

**Comment by liamcain** [comment]
- [Image Path](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/db8ffb818422d5b555cbb06102f46e46cdf575f9/src/main.ts#L227) setting names should use [Sentence case](https://en.wiktionary.org/wiki/sentence_case). This maximizes readability.
- [if(['md'].includes(fileName.split('.')[1])) {](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/db8ffb818422d5b555cbb06102f46e46cdf575f9/src/main.ts#L144) this function could be a lot simpler: `fileName.endsWith('.md')`
- [getAllLoadedFiles](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/db8ffb818422d5b555cbb06102f46e46cdf575f9/src/main.ts#L115) you should use [getFiles](https://github.com/obsidianmd/obsidian-api/blob/d6f80eccf6a6f0d1ec314605bdb7e9489311503f/obsidian.d.ts#L3830) instead of writing your own. You can also use [getMarkdownFiles](https://github.com/obsidianmd/obsidian-api/blob/d6f80eccf6a6f0d1ec314605bdb7e9489311503f/obsidian.d.ts#L3826) to automatically filter to md files
- [adapter.write](https://github.com/kaffarell/obsidian-tesseract-ocr/blob/db8ffb818422d5b555cbb06102f46e46cdf575f9/src/main.ts#L90) [Prefer the Vault API over the Adapter API](https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md#prefer-the-vault-api-over-the-adapter-api).

File: community-plugins.json [hunk 2]
```diff
@@ -6781,5 +6781,12 @@
     "author": "Dale de Silva",
     "description": "Allows import of Google Keep backup json files and their attachments. Can also be used to import other files.",
     "repo": "daledesilva/obsidian_google-keep-import"
-  }
+  },
+  {
+    "id": "image-ocr",
+    "name": "Image OCR",
+    "description": "Runs ocr on images and pastes result in description.",
```
Comment (liamcain) [inline community-plugins.json:6788]:
```suggestion
    "description": "Runs OCR on images and pastes result in description.",
```

**Pull request merged**